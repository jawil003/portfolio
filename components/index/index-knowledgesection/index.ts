export * from "./index-knowledgesection.test";
export * from "./index-knowledgesection-table/index-knowledgesection-table";