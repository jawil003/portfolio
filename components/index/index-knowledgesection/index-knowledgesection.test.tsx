import { render } from "@testing-library/react";
import { IndexKnowledgeSection } from "./index-knowledgesection";
import reactIntersectionObserver from "react-intersection-observer";
import { mocked } from 'ts-jest/utils';
import {createRef} from "react";

jest.mock("react-intersection-observer", () => ({
  useInView: jest.fn().mockResolvedValue({ref: {}})
}));

const mockedObserver = mocked(reactIntersectionObserver);

describe("IndexKnowledgeSection", () => {
  
  test("Create IndexKnowledgeSection", () => {
    const testIndexKnowledgeSection = render(
      <IndexKnowledgeSection
        items={[
          {
            title: "test",
            description: "",
            items: ["test", "test1"],
          },
        ]}
      />,
    );

    expect(
      testIndexKnowledgeSection.baseElement,
    ).toMatchSnapshot();
  });
});
