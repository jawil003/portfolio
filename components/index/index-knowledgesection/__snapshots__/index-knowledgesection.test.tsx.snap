// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexKnowledgeSection Create IndexKnowledgeSection 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: calc(30px / -2) calc(30px / -2);
}

.emotion-0>* {
  margin: calc(30px / 2) calc(30px / 2);
}

.emotion-1 {
  padding: 40px;
  min-height: 430px;
  width: 300px;
  background-color: #00739d;
  border-radius: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 40px;
}

.emotion-2 {
  text-align: start;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  letter-spacing: normal;
  word-spacing: normal;
  color: #f3f3f3;
  word-wrap: unset;
  padding: unset;
  word-break: normal;
}

.emotion-2 {
  font-weight: 400;
  font-size: 1.5em;
}

.emotion-4 {
  text-align: start;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  letter-spacing: normal;
  word-spacing: normal;
  color: #f3f3f3;
  word-wrap: unset;
  padding: unset;
  word-break: normal;
}

.emotion-4 {
  font-weight: 400;
  font-size: 0.875em;
}

.emotion-4 {
  margin: 0px;
}

.emotion-6 {
  padding-left: 20px;
}

.emotion-7 {
  color: #f3f3f3;
}

<div
  class="emotion-0"
  style="opacity: 0;"
>
  <div
    class="emotion-1"
  >
    <div>
      <h5
        class="emotion-2 emotion-3"
        color="#f3f3f3"
        id="heading5"
      >
        test
      </h5>
      <p
        class="emotion-4 emotion-5"
        color="#f3f3f3"
        id="body2"
      />
    </div>
    <ul
      class="emotion-6"
    >
      <li
        class="emotion-7"
      >
        test
      </li>
      <li
        class="emotion-7"
      >
        test1
      </li>
    </ul>
  </div>
</div>
`;
